bot/exts/github/_profile.py

```diff
@@ -0,0 +1,109 @@
+            if user_data["blog"].startswith("http"):  # Blog link is complete
+                blog = user_data["blog"]
+            elif user_data["blog"]:  # Blog exists but the link is not complete
+                blog = f"https://{user_data['blog']}"
+            else:
+                blog = "No website link available"
+
+            filtered_user_data["login"] = user_data["login"]
+            filtered_user_data["avatar"] = user_data["avatar_url"]
+            filtered_user_data["bio"] = (
+                f"```{user_data['bio']}```\n" if user_data["bio"] is not None else ""
+            )
+            filtered_user_data["created_at"] = user_data["created_at"]
+            filtered_user_data[
+                "Followers"
+            ] = f"[{user_data['followers']}]({user_data['html_url']}?tab=followers)"
+            filtered_user_data[
+                "Following"
+            ] = f"[{user_data['following']}]({user_data['html_url']}?tab=following)"
+            filtered_user_data["Public repos"] = (
+                f"[{user_data['public_repos']}]({user_data['html_url']}?tab"
+                f"=repositories) "
+            )
+            filtered_user_data[
+                "Gists"
+            ] = f"[{user_data['public_gists']}](https://gist.github.com/{username})"
+            filtered_user_data["Organizations"] = (
+                " | ".join(orgs) if orgs else "No Organizations"
+            )
+            filtered_user_data["Website"] = blog
+
+            return filtered_user_data
+
+    @staticmethod
+    def generate_embed(user_data: dict) -> discord.Embed:
+        """Generate Embed for github profile."""
+        embed = discord.Embed(
+            title=f"{user_data['login']}'s GitHub profile info",
+            description=user_data["bio"],
+            colour=discord.Colour.green(),
+            timestamp=datetime.strptime(user_data["created_at"], "%Y-%m-%dT%H:%M:%SZ"),
+        )
+        embed.set_thumbnail(url=user_data["avatar"])
+        embed.set_footer(text="Account created at")
+
+        embed.add_field(name="Followers", value=user_data["Followers"])
+        embed.add_field(name="\u200b", value="\u200b")
+        embed.add_field(name="Following", value=user_data["Following"])
+        embed.add_field(name="Public repos", value=user_data["Public repos"])
+        embed.add_field(name="\u200b", value="\u200b")
+        embed.add_field(name="Gists", value=user_data["Gists"])
+        embed.add_field(name="Organizations", value=user_data["Organizations"])
+        embed.add_field(name="\u200b", value="\u200b")
+        embed.add_field(name="Website", value=user_data["Website"])
+
+        return embed
+
+    async def get_github_info(self, username: Optional[str]) -> Union[None, Embed]:
+        """
+        Fetches a user's GitHub information.
+
+        Username is optional and sends the help command if not specified.
+        """
+        if username is None:
+            return None
```

Comment:
```suggestion
    async def get_github_info(self, username: str) -> Embed:
        """
        Fetches a user's GitHub information.
        """
```
Adjust annotations.

[2/2](https://github.com/gurkult/gurkbot/pull/66/files#diff-28432be4eec36ad2779e0c64d0500290051c943cddc99bec6900dfddafb39a7eR30)